From 76570ca1aec90fa2178c9f15cc2eeb63b80753d4 Mon Sep 17 00:00:00 2001
From: xplodwild <xplod@ouverta.fr>
Date: Mon, 23 Sep 2013 22:43:42 +0200
Subject: [PATCH] Setting for translucent statusbar on lockscreen

Change-Id: Id298c981c5c065d9d311fff8f71d140e702b4bc8

Conflicts:
	packages/SystemUI/res/values/colors.xml
	packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
	policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
---
 core/java/android/provider/Settings.java                 |  6 ++++++
 .../android/systemui/statusbar/phone/PhoneStatusBar.java | 16 +++++++++++++++-
 .../android/internal/policy/impl/PhoneWindowManager.java | 10 ++++++++++
 3 files changed, 31 insertions(+), 1 deletion(-)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 336ec16..99289da 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -3311,6 +3311,12 @@ public final class Settings {
         public static final String VOLUME_KEYS_CONTROL_RING_STREAM = "volume_keys_control_ring_stream";
 
         /**
+         * Whether or not to make the statusbar translucent on the keyguard
+         * @hide
+         */
+        public static final String STATUS_BAR_TRANSPARENT_ON_KEYGUARD = "status_bar_transparent_on_keyguard";
+
+        /**
          * Performance profile
          * @see config_perf_profile_prop in frameworks/base/core/res/res/values/config.xml
          * @hide
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
index fd7b67d..025c68b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
@@ -564,7 +564,7 @@ public class PhoneStatusBar extends BaseStatusBar {
         }
 
         // figure out which pixel-format to use for the status bar.
-        mPixelFormat = PixelFormat.OPAQUE;
+        updateTranslucentStatus();
 
         mSystemIconArea = (LinearLayout) mStatusBarView.findViewById(R.id.system_icon_area);
         mStatusIcons = (LinearLayout)mStatusBarView.findViewById(R.id.statusIcons);
@@ -2847,6 +2847,7 @@ public class PhoneStatusBar extends BaseStatusBar {
                 // work around problem where mDisplay.getRotation() is not stable while screen is off (bug 7086018)
                 repositionNavigationBar();
                 notifyNavigationBarScreenOn(true);
+                updateTranslucentStatus();
             }
         }
     };
@@ -2972,6 +2973,19 @@ public class PhoneStatusBar extends BaseStatusBar {
         mRecreating = false;
     }
 
+    public void updateTranslucentStatus() {
+        boolean translucent = false;
+        try {
+            translucent = mWindowManagerService.isBarTranslucent();
+        } catch (RemoteException ex) {
+            // do nothing
+        }
+        Log.e(TAG, "Translucent? " + translucent);
+        mPixelFormat = translucent ? PixelFormat.TRANSLUCENT : PixelFormat.OPAQUE;
+
+        mStatusBarView.getBackground().setAlpha(translucent ? 128 : 255);
+    }
+
     /**
      * Reload some of our resources when the configuration changes.
      *
diff --git a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
index ecd9489..d954621 100644
--- a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
+++ b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
@@ -5871,4 +5871,14 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 pw.print(" mDemoHdmiRotationLock="); pw.println(mDemoHdmiRotationLock);
         pw.print(prefix); pw.print("mUndockedHdmiRotation="); pw.println(mUndockedHdmiRotation);
     }
+
+    /** Translucent bars */
+    public boolean isBarTranslucent() {
+        return mBarsAreTranslucent && (Settings.System.getInt(mContext.getContentResolver(),
+                Settings.System.STATUS_BAR_TRANSPARENT_ON_KEYGUARD, 1) == 1);
+    }
+
+    public void setBarTranslucentAllowed(boolean allowed) {
+        mBarsAreTranslucent = allowed;
+    }
 }
-- 
1.8.4.2

